### PR TITLE
dcp-142 geo to hca - ui

### PR DESCRIPTION
**Pull request by amnonkhen**

### CODE REVIEW

File: src/app/shared/services/broker.service.ts [hunk 1]
```diff
@@ -75,4 +75,14 @@ export class BrokerService {
     };
   }
 
+  public downloadSpreadsheetUsingGeo(geoAccession: string) {
+    console.log("am in the downloadSpreadsheetUsingGeo function")
+    // const x = this.API_URL;
+    const x = 'http://127.0.0.1:5000'
```
Comment (amnonkhen) [inline src/app/shared/services/broker.service.ts:81]: I suggest to figure out why `this.API_URL` is not working.

File: src/app/shared/services/broker.service.ts [hunk 2]
```diff
@@ -75,4 +75,14 @@ export class BrokerService {
     };
   }
 
+  public downloadSpreadsheetUsingGeo(geoAccession: string) {
+    console.log("am in the downloadSpreadsheetUsingGeo function")
```
Comment (amnonkhen) [inline src/app/shared/services/broker.service.ts:79]: these log messages would have to eventually go way.

File: src/app/shared/services/broker.service.ts [hunk 3]
```diff
@@ -75,4 +75,27 @@ export class BrokerService {
...
+    const params = {
+      'accession': geoAccession,
+    };
+    return this.http
+      .post(`${ this.API_URL}/import-geo`, null,
+        {params, responseType: 'blob', observe: 'response'}).pipe(
+          map(response => {
+            if (response.status == HttpStatusCode.Ok) {
+              const contentDisposition = response.headers.get('content-disposition');
+              const filename = contentDisposition.split(';')[1].split('filename')[1].split('=')[1].trim();
```
Comment (aaclan-ebi) [inline src/app/shared/services/broker.service.ts:88]: could this be extracted to a function?

**Review by aaclan-ebi — APPROVED** [review]

File: src/app/projects/dialogs/geo-accession-dialog/geo-accession-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  onDownload() {
+    if (this.geoAccessionCtrl.value) {
+      this.loaderService.display(true, 'This may take a moment. Please wait...');
+      this.onClose();
+      this.brokerService.downloadSpreadsheetUsingGeo(this.geoAccessionCtrl.value)
+        .subscribe(response => {
+          const filename = response['filename'];
+          const blob = new Blob([response['data']]);
+          this.saveFileService.saveFile(blob, filename);
+          this.loaderService.display(false);
```
Comment (amnonkhen) [inline src/app/projects/dialogs/geo-accession-dialog/geo-accession-dialog.component.ts:43]: Boolean params are evil.
Have you considered `this.loaderService.hide()`?

File: src/app/projects/dialogs/geo-accession-dialog/geo-accession-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        },
+        error => {
+          this.alertService.error('Unable to download spreadsheet. Please retry later.', error.message );
```
Comment (amnonkhen) [inline src/app/projects/dialogs/geo-accession-dialog/geo-accession-dialog.component.ts:46]: Should the error message say something about geo? e.g. "Unable to read data from Geo"

File: src/app/shared/services/broker.service.ts (hunk 3 above)
Comment (amnonkhen) [reply src/app/shared/services/broker.service.ts:88]: 💯

**Comment by amnonkhen** [comment]
overall LGTM. A few minor comments that need to be addressed.

File: src/app/projects/dialogs/geo-accession-dialog/geo-accession-dialog.component.ts (hunk 4 above)
Comment (yusra-haider) [reply src/app/projects/dialogs/geo-accession-dialog/geo-accession-dialog.component.ts]: created hide() in loaderService

**Review by aaclan-ebi — APPROVED** [review]

**Review by ke4 — APPROVED** [review]
LGTM 💯

**Pull request merged**